### PR TITLE
fix(forge): new error message for vm.addr() when passing invalid private keys

### LINKED ISSUE #2158: Display a clearer error message when the private key passed to the `addr` cheatcode is greater than the secp256k1 curve order
### Component

Forge

### Describe the feature you would like

The order of the [secp256k1](https://en.bitcoin.it/wiki/Secp256k1) curve is a little bit smaller than `2^256`, which means that not all `uint256` values can be used as private keys. The order has this value:

```
0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141
```

Currently, when we pass a value greater than this to the `vm.addr` cheatcode, we got the following error:

> signature error

It would be nice to display a clearer, more specific error, to let users that what happened is that the private key is greater than what is permitted by secp256k1.

### Additional context

It is quite easy for a user to end up passing a value greater than the curve order:

1. Fuzz the private key as an `uint256` argument in a test function
2. Use the `vm.addr` cheatcode to generate an address from the fuzzed private key

### PR DESCRIPTION
**Pull request by 0xvv**
Fixes #2158 

I thought about storing the curve order value in another way, like an array of bytes, but i think the marginal perf gain is not worth the less readable code.

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/util.rs [hunk 1]
```diff
@@ -26,6 +27,13 @@ fn addr(private_key: U256) -> Result<Bytes, Bytes> {
         return Err("Private key cannot be 0.".to_string().encode().into())
     }
 
+    let secp256k1_order =
+        U256::from_str("0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141")
+            .unwrap();
+    if private_key > secp256k1_order {
+        return Err("Private key is greater than secp256k1 curve order.".to_string().encode().into())
```
Comment (mds1) [inline evm/src/executor/inspector/cheatcodes/util.rs:34]: I suspect most user's won't know what "curve order" means or what that value is. What do you think about: 
```suggestion
        return Err("Private key must be less than 115792089237316195423570985008687907852837564279074904382605163141518161494337.".to_string().encode().into())
```

File: evm/src/executor/inspector/cheatcodes/util.rs [hunk 2]
```diff
@@ -26,6 +27,13 @@ fn addr(private_key: U256) -> Result<Bytes, Bytes> {
         return Err("Private key cannot be 0.".to_string().encode().into())
     }
 
+    let secp256k1_order =
+        U256::from_str("0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141")
+            .unwrap();
+    if private_key > secp256k1_order {
```
Comment (mds1) [inline evm/src/executor/inspector/cheatcodes/util.rs:33]: We also need the same check in the `sign` method below 🙂

**Comment by 0xvv** [comment]
By the way does anyone know a dependency that re exports : https://docs.rs/secp256k1/latest/secp256k1/constants/constant.CURVE_ORDER.html

It would be the cleaner way to use the dep's constant

File: evm/src/executor/inspector/cheatcodes/util.rs [hunk 3]
```diff
@@ -26,6 +30,10 @@ fn addr(private_key: U256) -> Result<Bytes, Bytes> {
         return Err("Private key cannot be 0.".to_string().encode().into())
     }
 
+    if private_key > U256::from_big_endian(&Secp256k1::ORDER.to_be_bytes()) {
+        return Err("Private key must be less than 115792089237316195423570985008687907852837564279074904382605163141518161494337.".to_string().encode().into())
```
Comment (gakonst) [inline evm/src/executor/inspector/cheatcodes/util.rs:34]:
```suggestion
        return Err("Private key must be less than 115792089237316195423570985008687907852837564279074904382605163141518161494337 (the secp256k1 curve order).".to_string().encode().into())
```

File: evm/src/executor/inspector/cheatcodes/util.rs [hunk 4]
```diff
@@ -39,6 +47,10 @@ fn sign(private_key: U256, digest: H256, chain_id: U256) -> Result<Bytes, Bytes>
...
```
Comment (gakonst) [inline evm/src/executor/inspector/cheatcodes/util.rs:51]:
```suggestion
        return Err("Private key must be less than 115792089237316195423570985008687907852837564279074904382605163141518161494337 (the secp256k1 curve order).".to_string().encode().into())
```

**Review by gakonst — APPROVED** [review]

**Pull request merged**